src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -308,6 +308,9 @@ public int BinarySearch(T item)
         public int BinarySearch(T item, IComparer<T>? comparer)
             => BinarySearch(0, Count, item, comparer);
 
+        internal int BinarySearch<TValue>(TValue value, Func<T, TValue, int> comparer)
+            => SegmentedArray.BinarySearch(_items, value, comparer);
```

Comment:
❗ This failed to account for cases where `Capacity < Count`

Reply:
oooh..